src/representative.jl

```diff
@@ -9,17 +18,106 @@ used for each representative period. In addition, each representative period
 has an associated share that specifies how much of the total duration that
 is attributed to it.
 
+!!! note
+    - All representative periods must use the same type for the `TimeStructure`.
+    - If the field `period_share` is not specified, it assigns the same probability to each
+      representative period.
+    - It is possible that `sum(period_share)` is larger or smaller than 1. This can lead to
+      problems in your application. Hence, it is advised to scale it. We provide currently a
+      warning as it would correspond to a breaking change.
+    - If you include [`OperationalScenarios`](@ref) in your time structure, it is important
+      that you the scenarios are within the representative periods, and note the other way.
+
 ### Example
 ```julia
 # A year represented by two days with hourly resolution and relative shares of 0.7 and 0.3
-periods = RepresentativePeriods(2, 8760, [0.7, 0.3], [SimpleTimes(24, 1), SimpleTimes(24,1)])
+RepresentativePeriods(8760, [0.7, 0.3], [SimpleTimes(24, 1), SimpleTimes(24,1)])
+RepresentativePeriods(8760, [0.7, 0.3], SimpleTimes(24, 1))
+
+# A year represented by two days with hourly resolution and relative shares of 0.5
+RepresentativePeriods(2, 8760, SimpleTimes(24, 1))
+RepresentativePeriods(8760, [SimpleTimes(24, 1), SimpleTimes(24,1)])
 ```
 """
-struct RepresentativePeriods{S,T,OP<:TimeStructure{T}} <: TimeStructure{T}
-    len::Int
+struct RepresentativePeriods{S<:Duration,T,OP<:TimeStructure{T}} <:
+       TimeStructure{T}
+    len::Integer
```

Comment:
I think this should be `len::Int` as `Integer` is an abstract type. I am not sure this is critical for any performance, but I think it is good practice to avoid abstract types in structures. Constructors are good with `Integer`.

Reply:
I changed everything to `Integer` after discussions with you guys, but I am fine to move back to `Int`.

Reply:
That's a good point, but if we want to take care of type stability, I would prefer to parametrize it properly (e.g. `len::I` where `I<:Integer` to allow to use e.g. Int32 on 64 bit platforms or vice versa. Is the OP completely concrete btw?

Reply:
I decided to use `Int` in all composite types while keeping `Integer` in all constructors. This should work as far as I can see it for all cases.

OP should in this situation be concrete, at least we create with, _e.g._,

```julia
summer_day = SimpleTimes(24, 1);
winter_day = SimpleTimes(24, 1);
rep = RepresentativePeriods(2, 8760.0, [.5, .5], [summer_day, winter_day]);
```

a `RepresentativePeriods{Float64, Int64, SimpleTimes{Int64}}`.